flaml/automl/task.py

```diff
@@ -0,0 +1,184 @@
+from abc import ABC, abstractmethod, abstractstaticmethod
+from typing import Dict, Tuple
+
+# TODO: if your task is not specified in here, define your task as an all-capitalized word
+from typing import Union
+
+import pandas as pd
+import numpy as np
+
+SEQCLASSIFICATION = "seq-classification"
+MULTICHOICECLASSIFICATION = "multichoice-classification"
+TOKENCLASSIFICATION = "token-classification"
+
+SEQREGRESSION = "seq-regression"
+
+TS_FORECASTREGRESSION = (
+    "forecast",
+    "ts_forecast",
+    "ts_forecast_regression",
+)
+REGRESSION = ("regression", SEQREGRESSION, *TS_FORECASTREGRESSION)
+TS_FORECASTCLASSIFICATION = "ts_forecast_classification"
+TS_FORECASTPANEL = "ts_forecast_panel"
+TS_FORECAST = (
+    *TS_FORECASTREGRESSION,
+    TS_FORECASTCLASSIFICATION,
+    TS_FORECASTPANEL,
+)
+CLASSIFICATION = (
+    "binary",
+    "multiclass",
+    "classification",
+    SEQCLASSIFICATION,
+    MULTICHOICECLASSIFICATION,
+    TOKENCLASSIFICATION,
+    TS_FORECASTCLASSIFICATION,
+)
+RANK = ("rank",)
+SUMMARIZATION = "summarization"
+NLG_TASKS = (SUMMARIZATION,)
+NLU_TASKS = (
+    SEQREGRESSION,
+    SEQCLASSIFICATION,
+    MULTICHOICECLASSIFICATION,
+    TOKENCLASSIFICATION,
+)
+NLP_TASKS = (*NLG_TASKS, *NLU_TASKS)
+
+
+def get_classification_objective(num_labels: int) -> str:
+    if num_labels == 2:
+        objective_name = "binary"
+    else:
+        objective_name = "multiclass"
+    return objective_name
+
+
+class Task(ABC):
+    estimators = {}
+
+    def __init__(
+        self,
+        task_name: str,
+        X_train: Union[np.ndarray, pd.DataFrame],
+        y_train: Union[np.ndarray, pd.DataFrame, pd.Series],
+    ):
```

Comment:
The constructor here indicates that a Task object needs to know about X_train and y_train when it's constructed. 
The implementation indicates that no reference to the dataset is stored inside the Task object.
Why? What's the relation between a Task object and a dataset exactly?

Reply:
for example, at this stage we could infer it it's a binary or multi-category classification, and whether it's a univariate or panel regression, so the user wouldn't have the hassle of specifying that